aws/resource_aws_macie2_classification_job_test.go

```diff
@@ -0,0 +1,307 @@
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAwsMacie2ClassificationJob_WithTags(t *testing.T) {
+	var macie2Output macie2.DescribeClassificationJobOutput
+	resourceName := "aws_macie2_classification_job.test"
+	bucketName := os.Getenv("AWS_S3_BUCKET_NAME")
+	accountID := os.Getenv("AWS_ACCOUNT_ID")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      nil,
+		ErrorCheck:        testAccErrorCheck(t, macie2.EndpointsID),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsMacieClassificationJobconfigCompleteWithTags(bucketName, accountID, macie2.JobStatusRunning),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsMacie2ClassificationJobExists(resourceName, &macie2Output),
+					resource.TestCheckResourceAttr(resourceName, "job_type", macie2.JobTypeScheduled),
+					resource.TestCheckResourceAttr(resourceName, "job_status", macie2.JobStatusRunning),
+					testAccCheckResourceAttrRfc3339(resourceName, "created_at"),
+					testAccCheckResourceAttrRfc3339(resourceName, "last_run_time"),
+				),
+			},
+			{
+				Config: testAccAwsMacieClassificationJobconfigComplete(bucketName, accountID, macie2.JobStatusUserPaused),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsMacie2ClassificationJobExists(resourceName, &macie2Output),
+					resource.TestCheckResourceAttr(resourceName, "job_type", macie2.JobTypeScheduled),
+					resource.TestCheckResourceAttr(resourceName, "job_status", macie2.JobStatusUserPaused),
+					testAccCheckResourceAttrRfc3339(resourceName, "created_at"),
+					testAccCheckResourceAttrRfc3339(resourceName, "last_run_time"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAwsMacie2ClassificationJobExists(resourceName string, macie2Session *macie2.DescribeClassificationJobOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).macie2conn
+		input := &macie2.DescribeClassificationJobInput{JobId: aws.String(rs.Primary.ID)}
+
+		resp, err := conn.DescribeClassificationJob(input)
+
+		if err != nil {
+			return err
+		}
+
+		if resp == nil {
+			return fmt.Errorf("macie ClassificationJob %q does not exist", rs.Primary.ID)
+		}
+
+		*macie2Session = *resp
+
+		return nil
+	}
+}
+
+func testAccAwsMacieClassificationJobconfigNameGenerated(nameBucket, accountID, jobType string) string {
+	return fmt.Sprintf(`
+resource "aws_macie2_account" "test" {}
+
+resource "aws_macie2_classification_job" "test" {
+  job_type = "%s"
+  s3_job_definition {
+    bucket_definitions {
+      account_id = "%s"
+      buckets    = ["%s"]
+    }
+  }
```

Comment:
For quoted string values, we prefer using `%q` instead of `"%s"`. We also prefer using indexed format verbs, e.g. `%[1]q`